cmd/operator-sdk/new/cmd.go

```diff
@@ -225,6 +208,15 @@ func mustBeNewProject() {
 }
 
 func doGoScaffold() error {
+	if operatorType == projutil.OperatorTypeGo {
+		if len(fs.APIVersion) != 0 || len(fs.Kind) != 0 {
+			return fmt.Errorf("operators of type Go do not use --api-version or --kind")
+		}
+		if err := projutil.CheckRepo(repo); err != nil {
+			return err
+		}
+	}
```

Comment:
We already know that `operatorType` is Go here. Can this be moved back to `verifyFlags()`?